detect_secrets/core/secrets_collection.py

```diff
@@ -55,7 +90,11 @@ def scan_files(self, *filenames: str, num_processors: Optional[int] = None) -> N
         if not num_processors:
             num_processors = mp.cpu_count()
 
-        with mp.Pool(processes=num_processors) as p:
+        with mp.Pool(
+            processes=num_processors,
+            initializer=initialize_settings,
+            initargs=(get_settings(),),
```

Comment:
```suggestion
            initializer=configure_settings_from_baseline,
            initargs=(get_settings().json(),),
```

Instead of passing around Python objects (and repeating the logic found in `detect_secrets.settings.configure_settings_from_baseline`), does this work instead?

Or, if we wanted to optimize it even more, we can create the JSON representation once, and pass it into `initargs`.